packages/web/components/Modal/ModalProvider.js

```diff
@@ -47,25 +56,49 @@ export const ModalProvider = ({ children }) => {
 	const [state, dispatch] = useReducer(modalReducer, INITIAL_STATE);
 	const ModalComponent = getModalComponent(state.modal);
 
+	useEffect(() => {
+		if (ModalComponent && document) {
+			document.body.classList.add('modal-open');
+		}
+
+		return () => {
+			if (document) {
+				document.body.classList.remove('modal-open');
+			}
+		};
+	}, [ModalComponent]);
+
 	const openModal = useCallback(
-		(name, props = {}) => dispatch({ type: 'OPEN_MODAL', payload: { name, props } }),
+		(name, props = {}, modalProps = INITIAL_STATE.modalProps) =>
+			dispatch({ type: 'OPEN_MODAL', payload: { name, props, modalProps } }),
 		[],
 	);
 	const closeModal = useCallback(() => dispatch({ type: 'CLOSE_MODAL' }), []);
 
+	const getModalWrapper = () => {
+		if (!ModalComponent) return null;
+
+		const { modalProps } = state;
+
+		if (modalProps.customModal) {
+			return React.createElement(ModalComponent, { closeModal, ...state.props });
+		}
+
+		return (
+			<Modal data-testid="modal">
```

Comment:
**Minor:** acho que no test seria melhor mockar essa dependência, a menos que você esteja testando as funcionalidades de abertura / fechamento do modal

```
jest.mock('path-to-my-component', () => {
    return {
        __esModule: true,
        default: () => {
            return <div data-testid="">Mocked Modal</div>;
        }
    };
});
```

Reply:
To testando sim, principalmente porque inseri uma nova prop que permite não usar o wrapper do ModalProvider. Nesse caso o Modal do curador recebe o `closeModal`